deepmd/entrypoints/test.py

```diff
@@ -57,7 +60,13 @@ def test(
     RuntimeError
         if no valid system was found
     """
-    all_sys = expand_sys_str(system)
+    if datafile is not None:
+        all_sys = open(datafile, 'r')
+        all_sys = all_sys.read().splitlines()
+        datafile.close()
```

Comment:
`datafile` is `str`.

Reply:
```suggestion
        with open(datafile, 'r') as f:
            all_sys = f.read().splitlines()
```

Reply:
By the way, you should manually test your commit if there is no unit test.

Reply:
done